### ISSUE TITLE
systemd_unit resource not creating units anymore

### ISSUE THREAD

**Issue by fphilippon** [chef/chef#5324]
## Description

Following #5210 the systemd_unit resource seems not be able to create systemd units anymore.

## Chef Version

Chef 12.14.60

## Platform Version

CentOS 7.2
Chef: 12.14.60

## Replication Case

Creating systemd unit using the systemd_unit resource.

I suspect the new following method in lib/chef/provider/systemd_unit.rb to always return false:

```
def systemd_analyze_cmd
  @systemd_analyze_cmd ||= "#{systemd_analyze_path} verify %{path}"
end
```
Furthermore i did a test without the %{path} variable and the check works as expected (success when systemd unit file looks good and fail when it's not).

## Client Output
```

 * systemd_unit[jmx_exporter_kafka_521707.service] action create
  Recipe: <Dynamically Defined Resource>
    * file[/etc/systemd/system/jmx_exporter_kafka_521707.service] action create
      
      ================================================================================
      Error executing action `create` on resource 'file[/etc/systemd/system/jmx_exporter_kafka_521707.service]'
      ================================================================================
      
      Chef::Exceptions::ValidationFailed
      ----------------------------------
      Proposed content for /etc/systemd/system/jmx_exporter_kafka_521707.service failed verification #<Chef::Resource::File::Verification:0x000000071c5f00>
      
      Cookbook Trace:
      ---------------
      /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'
      
      Compiled Resource:
      ------------------
      # Declared in 
      
      file("/etc/systemd/system/jmx_exporter_kafka_521707.service") do
        action [:create]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        mode "0644"
        content "[Unit]\nDescription = jmx exporter\nAfter = network.target\n\n[Service]\nType = simple\nUser = prometheus\nGroup = prometheus\nRestart = on-failure\nExecStart = /usr/bin/java -Dcom.sun.management.jmxremote.ssl=false   -Dcom.sun.management.jmxremote.authenticate=false   -Dcom.sun.management.jmxremote.port=5555 -jar                 /opt/jmx_exporter/jmx-exporter.jar                 5556                 /opt/jmx_exporter/kafka_521707.yml\n\n[Install]\nWantedBy = multi-user.target\n"
        path "/etc/systemd/system/jmx_exporter_kafka_521707.service"
        owner "root"
        group "root"
        verifications [#<Chef::Resource::File::Verification:0x000000071c5f00 @command_opts={}, @command="/bin/systemd-analyze verify %{path}", @block=nil, @parent_resource=<file[/etc/systemd/system/jmx_exporter_kafka_521707.service] @name: "/etc/systemd/system/jmx_exporter_kafka_521707.service" @noop: nil @before: nil @params: {} @provider: nil @allowed_actions: [:nothing, :create, :delete, :touch, :create_if_missing] @action: [:create] @updated: false @updated_by_last_action: false @supports: {} @ignore_failure: false @retries: 0 @retry_delay: 2 @source_line: nil @guard_interpreter: nil @default_guard_interpreter: :default @elapsed_time: 0 @sensitive: false @mode: "0644" @content: "[Unit]\nDescription = jmx exporter\nAfter = network.target\n\n[Service]\nType = simple\nUser = prometheus\nGroup = prometheus\nRestart = on-failure\nExecStart = /usr/bin/java -Dcom.sun.management.jmxremote.ssl=false   -Dcom.sun.management.jmxremote.authenticate=false   -Dcom.sun.management.jmxremote.port=5555 -jar                 /opt/jmx_exporter/jmx-exporter.jar                 5556                 /opt/jmx_exporter/kafka_521707.yml\n\n[Install]\nWantedBy = multi-user.target\n" @path: "/etc/systemd/system/jmx_exporter_kafka_521707.service" @owner: "root" @group: "root" @verifications: [...]>>]
      end
      
      Platform:
      ---------
      x86_64-linux
      
    
    ================================================================================
    Error executing action `create` on resource 'systemd_unit[jmx_exporter_kafka_521707.service]'
    ================================================================================
    
    Chef::Exceptions::ValidationFailed
    ----------------------------------
    file[/etc/systemd/system/jmx_exporter_kafka_521707.service] (dynamically defined) had an error: Chef::Exceptions::ValidationFailed: Proposed content for /etc/systemd/system/jmx_exporter_kafka_521707.service failed verification #<Chef::Resource::File::Verification:0x000000071c5f00>
    
    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'
    
    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/prometheus-platform/recipes/jmx_exporter.rb
    
     90:         systemd_unit "jmx_exporter_#{target['name']}.service" do
     91:           enabled true
     92:           active true
     93:           masked false
     94:           static false
     95:           content unit
     96:           triggers_reload true
     97:           action [:create, :enable, :start]
     98:         end
     99:       end
    
    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/prometheus-platform/recipes/jmx_exporter.rb:90:in `block in from_file'
    
    systemd_unit("jmx_exporter_kafka_521707.service") do
      action [:create, :enable, :start]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :systemd_unit
      cookbook_name "prometheus-platform"
      recipe_name "jmx_exporter"
      enabled true
      active true
      content {"Unit"=>{"Description"=>"jmx exporter", "After"=>"network.target"}, "Service"=>{"Type"=>"simple", "User"=>"prometheus", "Group"=>"prometheus", "Restart"=>"on-failure", "ExecStart"=>"/usr/bin/java -Dcom.sun.management.jmxremote.ssl=false   -Dcom.sun.management.jmxremote.authenticate=false   -Dcom.sun.management.jmxremote.port=5555 -jar                 /opt/jmx_exporter/jmx-exporter.jar                 5556                 /opt/jmx_exporter/kafka_521707.yml"}, "Install"=>{"WantedBy"=>"multi-user.target"}}
      triggers_reload true
    end
    
    Platform:
    ---------
    x86_64-linux
    

Running handlers:
  - Chef::Handler::Slack
Running handlers complete
Chef Client failed. 5 resources updated in 15 seconds
```

**Comment by s-bernard** [comment]
It seems to me that (at least on Centos 7.2) "systemd-analyze verify" does not accept a filename not ending by an unit extension like "service", "mount", "target", which is not the case of tempfiles.

**Comment by lamont-granquist** [comment]
#5327 should fix this and should fix this whole class of problems with verifiers and extensions.   i don't think the extension on the tempfile was a public api so just fixed it there.

**Comment by nathwill** [comment]
this doesn't seem to be resolved in 12.14.77; i'm still seeing `Chef::Exceptions::ValidationFailed` for any `systemd_unit` resources.

**Comment by nathwill** [comment]
correction, not *any*, but specific unit types where the name matters relative to the purpose: e.g. automount, mount, device, etc. opening a separate issue for this, as it's a slightly different problem.


```bash
[root@default-fedora-24 ~]# systemd-analyze verify /etc/systemd/system/chef-proc-sys-fs-binfmt_misc.automount 
chef-proc-sys-fs-binfmt_misc.automount: Where= setting doesn't match unit name. Refusing.
chef-proc-sys-fs-binfmt_misc.automount: Failed to create chef-proc-sys-fs-binfmt_misc.automount/start: Unit chef-proc-sys-fs-binfmt_misc.automount is not loaded properly: Invalid argument.
```
